### ISSUE TITLE
"tensorflow op FusedBatchNorm is not supported" on ResNet V2 152 Model

### ISSUE THREAD

**Issue by pengwa** [onnx/tensorflow-onnx#65]
ResNet V1 152 Model can be converted successfully, V2 152 is not. 

The reason is that: 

1. in V1 152, FusedBatchNorm are used in following pattern
```
node {
  name: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/BatchNorm/FusedBatchNorm"
  op: "FusedBatchNorm"
  input: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/Conv2D"
  input: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/BatchNorm/gamma/read"
  input: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/BatchNorm/beta/read"
  input: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/BatchNorm/moving_mean/read"
  input: "resnet_v1_152/block4/unit_2/bottleneck_v1/conv3/BatchNorm/moving_variance/read"
 ...
},
```
https://github.com/tensorflow/tensorflow/blob/4bd6ac7ce02338cf6d0cc6c7ecfe4bd786d45e1e/tensorflow/tools/graph_transforms/fold_old_batch_norms.cc can help convert FusedBatchNorm into few other smaller operators with pattern:
```
 {"BatchNormWithGlobalNormalization|FusedBatchNorm",    // batch_norm_node
        {
          {"Conv2D",                          // conv_node
            {
              {"*"},                          // input_node
              {"Const"},                      // weights_node
            }
          },
          {"Const"},                          // mean_node
          {"Const"},                          // variance_node
          {"Const"},                          // beta_node
          {"Const"},                          // gamma_node
        }
      }
```

2. in V2 152, besides the above usage pattern, there are also others:

```
node {
  name: "resnet_v2_152/block1/unit_1/bottleneck_v2/preact/FusedBatchNorm"
  op: "FusedBatchNorm"
  input: "resnet_v2_152/pool1/MaxPool"
  input: "resnet_v2_152/block1/unit_1/bottleneck_v2/preact/gamma/read"
  input: "resnet_v2_152/block1/unit_1/bottleneck_v2/preact/beta/read"
  input: "resnet_v2_152/block1/unit_1/bottleneck_v2/preact/moving_mean/read"
  input: "resnet_v2_152/block1/unit_1/bottleneck_v2/preact/moving_variance/read"
...
}

node {
  name: "resnet_v2_152/block1/unit_2/bottleneck_v2/preact/FusedBatchNorm"
  op: "FusedBatchNorm"
  input: "resnet_v2_152/block1/unit_1/bottleneck_v2/add"
  input: "resnet_v2_152/block1/unit_2/bottleneck_v2/preact/gamma/read"
  input: "resnet_v2_152/block1/unit_2/bottleneck_v2/preact/beta/read"
  input: "resnet_v2_152/block1/unit_2/bottleneck_v2/preact/moving_mean/read"
  input: "resnet_v2_152/block1/unit_2/bottleneck_v2/preact/moving_variance/read"
...
}
```

**Comment by pengwa** [comment]
tried latest master branch, the issue is fixed. 


-------------------- start handling resnet_v2_152_inf_graph.pb ----------------------------
change working directory to /home/pengwang/community/tensorflow
------ summarize the frozen graph, to get the inputs and outputs name
bazel-bin/tensorflow/tools/graph_transforms/summarize_graph --in_graph=/tmp/frozen/resnet_v2_152_inf_graph.pb
Found 1 possible inputs: (name=input, type=float(1), shape=[?,224,224,3])
No variables spotted.
Found 1 possible outputs: (name=resnet_v2_152/predictions/Reshape_1, op=Reshape)
Found 60382733 (60.38M) const parameters, 0 (0) variable parameters, and 0 control_edges
Op types used: 822 Const, 816 Identity, 156 Conv2D, 151 FusedBatchNorm, 151 Relu, 56 BiasAdd, 50 Add, 4 MaxPool, 4 Pad, 2 Reshape, 1 Mean, 1 Placeholder, 1 Shape, 1 Softmax, 1 Squeeze
To use with tensorflow/tools/benchmark:benchmark_model try these arguments:
bazel run tensorflow/tools/benchmark:benchmark_model -- --graph=/tmp/frozen/resnet_v2_152_inf_graph.pb --show_flops --input_layer=input --input_layer_type=float --input_layer_shape=-1,224,224,3 --output_layer=resnet_v2_152/predictions/Reshape_1
------ update the inputs and outputs name to format like input_name:index
python3 /home/pengwang/community/learning/onnx/update_name_with_index.py input
updated input names is input:0, output names is resnet_v2_152/predictions/Reshape_1:0
------ start convertion, tensorflow usage require caller program must not in tensorflow root folder, so switch to current user directory with cd
using tensorflow=1.9.0-rc0, onnx=1.2.2
2018-07-10 15:45:51.075989: I tensorflow/tools/graph_transforms/transform_graph.cc:318] Applying fold_batch_norms
2018-07-10 15:45:51.343881: I tensorflow/tools/graph_transforms/transform_graph.cc:318] Applying fold_old_batch_norms
2018-07-10 15:45:52.718025: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2018-07-10 15:45:52.796077: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties:
name: Quadro K620 major: 5 minor: 0 memoryClockRate(GHz): 1.124
pciBusID: 0000:03:00.0
totalMemory: 1.95GiB freeMemory: 1.82GiB
2018-07-10 15:45:52.796109: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2018-07-10 15:45:52.932926: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-07-10 15:45:52.932964: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0
2018-07-10 15:45:52.932970: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N
2018-07-10 15:45:52.933090: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 1576 MB memory) -> physical GPU (device: 0, name: Quadro K620, pci bus id: 0000:03:00.0, compute capability: 5.0)
tensorflow ops: Counter({'Const': 822, 'Identity': 816, 'Conv2D': 156, 'FusedBatchNorm': 151, 'Relu': 151, 'BiasAdd': 56, 'Add': 50, 'Pad': 4, 'MaxPool': 4, 'Reshape': 2, 'Mean': 1, 'Shape': 1, 'Softmax': 1, 'Placeholder': 1, 'Squeeze': 1})
tensorflow attr: Counter({'T': 1394, 'dtype': 823, 'value': 822, '_class': 816, 'data_format': 367, 'strides': 160, 'padding': 160, 'use_cudnn_on_gpu': 156, 'dilations': 156, 'is_training': 151, 'epsilon': 151, 'Tpaddings': 4, 'ksize': 4, 'Tshape': 2, 'out_type': 1, 'Tidx': 1, 'squeeze_dims': 1, 'shape': 1, 'keep_dims': 1})
onnx mapped: Counter({'Const': 822, 'Identity': 816, 'Conv2D': 156, 'FusedBatchNorm': 151, 'Relu': 151, 'BiasAdd': 56, 'Add': 50, 'Pad': 4, 'MaxPool': 4, 'Reshape': 2, 'Mean': 1, 'Shape': 1, 'Softmax': 1, 'Placeholder': 1, 'Squeeze': 1})
onnx unmapped: Counter()

**Comment by pengwa** [comment]
it should be fixed by commit: https://github.com/onnx/tensorflow-onnx/commit/05ee1a7ce1b63a90794f3d64addb45ae913ce84e, all FusedBatchNorm tensorflow operators are mapped to BatchNormalization.
